include/NeoFOAM/finiteVolume/cellCentred/operators/fvccSparsityPattern.hpp

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2025 NeoFOAM authors
+
+#pragma once
+
+#include "NeoFOAM/mesh/unstructured.hpp"
+#include "NeoFOAM/linearAlgebra.hpp"
+
+namespace la = NeoFOAM::la;
+
+namespace NeoFOAM::finiteVolume::cellCentred
+{
+
+
+class SparsityPattern
+{
+public:
+
+    SparsityPattern(const UnstructuredMesh& mesh);
+
+    void update();
+
+    const la::LinearSystem<NeoFOAM::scalar, NeoFOAM::localIdx>& linearSystem() const;
+
+    const Field<uint8_t>& ownerOffset() const;
+
+    const Field<uint8_t>& neighbourOffset() const;
+
+    const Field<uint8_t>& diagOffset() const;
+
+    // add selection mechanism via dictionary later
```

Comment:
use TODO: we can find things like that later.

```suggestion
    // TODO: add selection mechanism via dictionary later
```